tests/cut_plot_test.py

```diff
@@ -88,6 +88,12 @@ def test_object_clicked(self, quick_options_mock):
     def test_update_legend(self):
         line = Line2D([], [])
         self.axes.get_legend_handles_labels = MagicMock(return_value=([line], ['some_label']))
+        self.cut_plot._legends_shown = False
```

Comment:
Good unit test, but typically we want to separate out each test case. For example `def test_update_legend_legends_shown` and `def test_update_legend_legends_not_shown`.

The title of the failure should point you directly to the failed case. This is not always the case within mantid and mslice, but for new tests we try to follow this.

Reply:
Ah I see! Will do that then.

Reply:
How does it look now?

Reply:
Thanks for separating them out.

Do you know what we are checking for with  `self.assertTrue(self.cut_plot._legends_visible[0])`? It is asserting `True` for both, intuitively it feels like it should be asserting `False` if the legends are not shown.

Reply:
Actually, I am not sure. There are 3 atributes of CutPlot that control the legend: `._lines_visible = {}`, .`_legends_visible = [] `and `._legends_shown = True`. My best guess thus far is that `_legends_visible` is a list of booleans that controls which lines are displayed in the legend box itself, rather than the whether the legend box is visible or not. 

Should I look into it further to make sure?

Reply:
That makes sense, it's probably work a check to confirm that, and maybe updating the variable name to something that is clearer!